Add variable truth tutorial

yymao opened the pull request:
This PR adds the variable truth tutorial and fixes #89. 

Feedback welcome.  There's a few cells that need to be removed before merging; I left them there for now so that the reviewer can run this notebook on NERSC more easily (you will need to use the `u/yymao/pr587-review` branch of `gcr-catalogs` for this notebook to work).

yymao commented:
I have updated this tutorial notebook to match the latest truth catalogs. It's now ready for review @JoanneBogart @wmwv

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+    "\n",
+    "**Learning objectives**: After going through this notebook, you should be able to:\n",
+    "  1. Know the difference of the various truth catalogs related variable truth information.\n",
+    "  2. Being able to load the light curves for any SNe or stars that are of interset. "
```
omit "being able to".   For 2. you just need "Load the light curves ..."

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "source": [
+    "## Supernova light curves\n",
+    "\n",
+    "We will start with finding light curves for supernovae.  When accessing DC2 variable truth information, one important concept is that the light curves are stored separetely from a \"summary table\". \n",
```
separetely --> separately

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+    "\n",
+    "We will start with finding light curves for supernovae.  When accessing DC2 variable truth information, one important concept is that the light curves are stored separetely from a \"summary table\". \n",
+    "\n",
+    "In the summary table, each row corresponds to one object (supernova), and it contains \"summary\" truth information such as coordinates, redshift, and time-averaged fluxes, but not the light curves. \n",
```
Maybe add a sentence like
"Here _summary_ implies associated with the source generally, not with a particular observation of the source." 
or modify the sentence you have:

In the summary table, each row corresponds to one object (here a supernova), and it contains _summary_ truth information such as coordinates, redshift, and time-averaged fluxes - that is, attributes associated with the object overall, not with a particular observation of it.  Hence it does not include light curve values.

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now actually load the data into memory. For this tutorial, we will only need the ID (`id`), coordinates (`ra`, `dec`), redshift. \n",
```
Replace comma before the word "redshift" with the word "and"

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have the summary table, we can use it to select, say supernovae near a certain coordinate or in a certain redshift range. In this tutorial, we will find all supernovae that are within 2 arcmin of (RA, Dec) = (60.5, -36.6) and within z < 0.8. We will store these supernovae in `objects_to_plot` which we will use later to extract and plot the their light curves."
```
Suggest ending sentence with
"all supernovae that are within 2 arcmin of (RA, Dec) = (60.5, -36.6) and satisfy z < 0.8."

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unlike the summary table, in the variability table, each row corresponds to one \"measurement\" the flux of a given object, at a given time, with a given filter. Hence, a single supernovae may correspond to many rows. \n",
```
1. Should not be a comma after variability table.  2. There is a missing word ("of") or missing punctuation (colon) after "measurement".   3. "corresponds" is maybe not the best word here.   I suggest

Unlike the summary table, in the variability table each row corresponds to one "measurement": the flux of a given object, at a given time, with a given filter. Hence, a single supernova is usually represented by many rows.

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can see that from above that the first five rows are all different measurements (at different time or with different filters) for a single object (`MS_9177_2005`). \n",
```
Suggest
You can see from the above the the first five rows are all different measurements (at different times and not always with the same filter) for a single...

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
````diff
@@ -0,0 +1,520 @@
+    "```python\n",
+    "GCRQuery((lambda x: np.isin(x, objects_to_plot[\"id\"].values), \"id\"))\n",
+    "```\n",
+    "Even with this filter, the cell below will take a while to run. If runtime is a concern, you may want to use the SQL database that we also provide to access these information instead."
````
Change "these information" to "this information"

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we have the light curves! You might notice that we used `delta_flux` to get the magnidue. Here `delta_flux` is the flux difference to the inifinite time averaged flux value. For SNe, the infinite time-averaged fluxes are always zero, and hence `delta_flux` would just store the actual flux values. As we will see later, for stars we will have to add the infinite time-averaged fluxes to `de ...
```
magnidue -->.  magnitude

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "source": [
+    "Here the fluxes (`flux_*`) in the stars' summary table are time-averaged fluxes over *infinite* time, which we will need when we plot the light curves.\n",
+    "\n",
+    "Also, not all stars in this summary table are \"variable\"; some are rather static. The column `above_threshold` tells us if the star is \"variable enough\" to be included in the variability table. So here we will also grab it when we are looking for interseting variable stars. "
```
interseting  -->. interesting

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Another important difference between stars and supernovae is that for stars, their infinite time-averaged fluxes are usualy non-zero! We can verify that:"
```
usualy --> usually

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Again, recall that in the \"variability table\", each row corresponds to one measurement, and we need to collect all the rows that have `id` values overlap with `objects_to_plot[\"id\"].values`. \n",
```
overlap    --> overlapping

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "source": [
+    "Again, recall that in the \"variability table\", each row corresponds to one measurement, and we need to collect all the rows that have `id` values overlap with `objects_to_plot[\"id\"].values`. \n",
+    "\n",
+    "Also remember that the variability table is a large table (even larger in the star case). We will use the same `filters` trick again to reduce the memory footprint, but the cell below will still result in ~4 GB at peak memory usage. It will take a while to run. If runtime is a concern, you may want to use the SQL database that we also provide to access these information instead."
```
these information  --> this information

JoanneBogart commented on notebooks/dc2/truth_variable_info.ipynb:
```diff
@@ -0,0 +1,520 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once we collect the light curve infomation in `d_lc`. We can use the same `groupby` trick to split the data frame by ID and bandpass, and proceed to make light curve plot."
```
Add "a" or "the" before "light curve plot".    E.g.,
..and proceed to make the light curve plot

JoanneBogart reviewed (changes requested):
Generally looks good!  I've made a number of minor suggestions - fixing typos, making the text more idiomatic or, in one or two cases, trying for more clarity.  Please also consider whether there is a way to make the variability in the  first star plot more evident.

yymao commented:
@JoanneBogart thanks for the review. I've addressed all the comments. You can see the detail changed in commit e8e087d. Regarding making the variability in the first star plot more evident: the second variability plot (difference in magnitude) is meant to address the issue. So instead of changing the plot, I just updated the text to make this point more explicit. 

I've also tested the truth_match notebook and everything works fine. Numpy is deprecating the use `np.int` in favor of `np.int64` so I made that change in cba6c3d.

Ready for re-review.

JoanneBogart reviewed (approved):
LGTM.
